eucadeploy/plugins/debugger/eucalyptus_sosreports.py

```diff
@@ -0,0 +1,150 @@
+                else:
+                    self.warning(host + ':Package not installed - '
+                                + package + '; Installing ' + package)
+                    with hide('everything'):
+                        yum_output = self.run_command_on_host('yum install '
+                                     + ' --assumeyes --quiet --nogpgcheck ' + package,
+                                     host=host)
+                    if not yum_output:
+                        self.success(host + ':Package installed - ' + package)
+                    else:
+                        self.failure(host + ':Package failed to install - ' + package)
+
+    def _grab_sosreports(self, all_hosts):
+        """
+        Create local directory
+        for downloaded sosreports
+        """
+        timestamp = datetime.strftime(datetime.now(), '%Y%m%d-%H%M%S') 
+        directory = 'sosreport-' + timestamp
+        message = 'Creating directory ' + directory + " on localhost"
+        self.info(message)
+        with hide('everything'):
+            mkdir_output = self.create_localdir(directory)
+        if re.search('created', mkdir_output):
+            self.success('localhost: Directory ' + directory
+                         + ' created successfully')
+        else:
+            self.failure('localhost: Directory ' + directory
+                         + ' creation failed')
+        """
+        Remove old sosreports are around in /tmp
+        """
+        with hide('everything'):
+            rm_output = self.run_command_on_hosts('rm -rf /tmp/sosreport*',
+                                                  hosts=all_hosts)
+        for host in all_hosts:
+            if not rm_output[host]:
+                self.success(host + ':old sosreports successfully removed')
+            else:
+                self.failure(host + ':old sosreports failed to be removed')
+
+        with hide('everything'):
+            """
+            Execute sosreport on all hosts
+            """
+            sosreport_output = self.execute_sosreports_on_hosts(hosts=all_hosts)
+        for host in all_hosts:
+            """
+            Confirm sosreport ran successfully;
+            Download sosreport to local client
+            """
+            hostname = host.replace(".", '')
+            sosfile = 'sosreport-' + hostname
+            for output in sosreport_output[host].split('\n'):
+                sosreport_file = re.search(sosfile, output, re.I)
+                if sosreport_file:
+                    self.success(host + ':sosreport finished - '
+                                + output)
+                    remote_path = output.strip()
+                    local_file = remote_path.split('/')[2]
+                    format_host = host.replace(".", "_")
+                    local_path = directory + "/" + format_host + "-" + local_file
+                    with hide('everything'):
+                        cp_output = self.get_file_on_host(remote_path,
```

Comment:
Did a quick test.  I removed the sosreport before the I was able to download it to my local client.  The following result was returned:
```
...
[INFO                ] Running sosreport on 10.111.1.127[INFO                ] Running sosreport on 10.111.1.111

[INFO                ] Running sosreport on 10.111.1.109
[INFO                ] Running sosreport on 10.111.5.111
[INFO                ] Running sosreport on 10.111.5.84
[INFO                ] Running sosreport on 10.111.5.114
[DEBUG PASSED        ] 10.111.5.114:sosreport finished -   /tmp/sosreport-101115114.000-20150518102624-9f8e.tar.xz

Fatal error: get() encountered an exception while downloading '/tmp/sosreport-101115114.000-20150518102624-9f8e.tar.xz'

Underlying exception:
    No such file

Aborting.
get() encountered an exception while downloading '/tmp/sosreport-101115114.000-20150518102624-9f8e.tar.xz'

Underlying exception:
    No such file
[DEBUG FAILED        ] 10.111.5.114:sosreport failed to download - sosreport-20150518-132526/10_111_5_114-sosreport-101115114.000-20150518102624-9f8e.tar.xz
[DEBUG PASSED        ] 10.111.5.84:sosreport finished -   /tmp/sosreport-10111584.000-20150518102557-5457.tar.xz
[DEBUG PASSED        ] 10.111.5.84:sosreport downloaded - sosreport-20150518-132526/10_111_5_84-sosreport-10111584.000-20150518102557-5457.tar.xz
[DEBUG PASSED        ] 10.111.5.111:sosreport finished -   /tmp/sosreport-101115111.000-20150518102711-0178.tar.xz
[DEBUG PASSED        ] 10.111.5.111:sosreport downloaded - sosreport-20150518-132526/10_111_5_111-sosreport-101115111.000-20150518102711-0178.tar.xz
[DEBUG PASSED        ] 10.111.1.127:sosreport finished -   /tmp/sosreport-101111127.000-20150518103116-319d.tar.xz
[DEBUG PASSED        ] 10.111.1.127:sosreport downloaded - sosreport-20150518-132526/10_111_1_127-sosreport-101111127.000-20150518103116-319d.tar.xz
[DEBUG PASSED        ] 10.111.1.111:sosreport finished -   /tmp/sosreport-101111111.000-20150518102635-ac50.tar.xz
....
```
If the file is not there, it will return an error message, then continue proceed.  I don't like how it spews crap to the screen. I will look at a cleaner way of handling this error.